### PR TITLE
Implement table updates by inserting into write buffer

### PR DESCRIPTION
**Pull request by mheinzel**
It might make sense to pull more parts of the table handle outside of the `RWVar` and sum type.

### CODE REVIEW

File: src/Database/LSMTree/Normal.hs [hunk 1]
```diff
@@ -251,7 +254,15 @@ updates ::
   => V.Vector (k, Update v blob)
   -> TableHandle m k v blob
   -> m ()
-updates = undefined
+updates es (TableHandle th) = do
+    -- make sure not to insert any blobs into monoidal tables
+    when (isJust (Internal.confResolveMupsert (Internal.tableConfig th))) $
+      throwIO Internal.ErrExpectedNormalTable
+    Internal.updates Internal.constResolveMupsert (V.mapStrict serialiseEntry es) th
```
Comment (jorisdral) [inline src/Database/LSMTree/Normal.hs:261]: I think passing `Internal.constResolveMupsert` here is not necessary: we already checked that `Internal.confResolveMupsert` is `Nothing,` and `Internal.updates` can read the `confResolveMupsert` from the table handle using `resolveMupsert`

**Review by jorisdral — COMMENTED** [review]
Looks good! There's only a minor thing I think we should change, but we can merge after that

> It might make sense to pull more parts of the table handle outside of the `RWVar` and sum type.

Let's do that on demand

File: src/Database/LSMTree/Normal.hs (hunk 1 above)
Comment (mheinzel) [reply src/Database/LSMTree/Normal.hs:261]: Ah, yes. It's redundant.

File: src/Database/LSMTree/Internal.hs [hunk 2]
```diff
@@ -530,13 +531,25 @@ toNormalLookupResult = \case
...
+updates es th = withOpenTable th $ \thEnv -> do
+    -- A placeholder implementation that is sufficient to pass the tests, but
+    -- keeps all entries in memory.
+    -- TODO: flush write buffer when full
+    -- TODO: merge runs when level becomes full
+    modifyMVar_ (tableWriteBuffer thEnv) $ \wb -> do
+      return $ foldl' (flip (uncurry (WB.addEntry resolve))) wb es
+  where
+    resolve =
+      maybe const unResolveMupsert (confResolveMupsert (tableConfig th))
```
Comment (jorisdral) [inline src/Database/LSMTree/Internal.hs:552]: You can use https://github.com/IntersectMBO/lsm-tree/pull/251/files#diff-b75c6514218a4064abdf1339bbf6a6e163f0be909f3d5cb7602dbfe6312f6023R660-R661 here

**Review by jorisdral — APPROVED** [review]

**Pull request merged**